modules/apps/dispatch/dispatch-web/src/main/java/com/liferay/dispatch/web/internal/display/context/DispatchTriggerDisplayContext.java

```diff
@@ -225,11 +229,24 @@ public SearchContainer<DispatchTrigger> getSearchContainer() {
 		_searchContainer.setOrderByComparator(null);
 		_searchContainer.setOrderByType(getOrderByType());
 		_searchContainer.setResultsAndTotal(
-			() -> _dispatchTriggerLocalService.getDispatchTriggers(
-				dispatchRequestHelper.getCompanyId(),
-				_searchContainer.getStart(), _searchContainer.getEnd()),
-			_dispatchTriggerLocalService.getDispatchTriggersCount(
-				dispatchRequestHelper.getCompanyId()));
+			ListUtil.filter(
+				_dispatchTriggerLocalService.getDispatchTriggers(
+					dispatchRequestHelper.getCompanyId(), QueryUtil.ALL_POS,
```

Comment:
@pei-jung `QueryUtil.ALL_POS` is wrong, please make it consistent with before by using `_searchContainer.getStart(), _searchContainer.getEnd()`

Reply:
This is how [searchContainer.setResultsAndTotal(List results)](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/dao/search/SearchContainer.java#L488-L492) works. It expects the entire results set in order to get the total count. The results list will be paginated with the start and end values.

Reply:
@pei-jung So it means "_searchContainer.getStart(), _searchContainer.getEnd()" is wrong? I am confused here, which part of logic make it consistent with "_searchContainer.getStart(), _searchContainer.getEnd()"?

Reply:
@tinatian the previous code calls `setResultsAndTotal(UnsafeSupplier<List<R>, E> setResultsUnsafeSupplier, int total)`, which sets the `results` and `totoal` fields separately. It calls `_dispatchTriggerLocalService.getDispatchTriggers(dispatchRequestHelper.getCompanyId(), _searchContainer.getStart(), _searchContainer.getEnd())` to get the paginated list for the table display and `_dispatchTriggerLocalService.getDispatchTriggersCount` for the total count. 

However, in order to apply the feature flag filter, we need to get all the dispatch triggers first, filter them by feature flag, then paginate it. We'll be calling the same service for the paginated list and total count, so I used `searchContainer.setResultsAndTotal(List results)` to save a service call. It still uses `_searchContainer.getStart(), _searchContainer.getEnd()` internally to set the paginated list. 

```
_setResults(results.subList(_start, _resultEnd));
```

Reply:
@pei-jung Thanks. I see... the API method name is so confusing here, you have to go down into its impl to tell the difference.